### PR TITLE
chore: improving naming from validator to authorizer

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

File: src/app/authentication/login.ts [hunk 1]
```diff
@@ -362,12 +364,15 @@ export const loginWithDevice = async ({
...
+    const authorizedIPMetadata = IPMetadataAuthorizer(
       ipMetadataValidationSettings,
-    ).validate(ipFetcherInfo)
+    ).authorize(ipFetcherInfo)
 
-    if (validatedIPMetadata instanceof Error) {
-      return new InvalidIPForOnboardingError(validatedIPMetadata.name)
+    if (authorizedIPMetadata instanceof Error) {
+      if (authorizedIPMetadata instanceof MissingIPMetadataError)
+        return new InvalidIpMetadataError(authorizedIPMetadata.name)
```
Comment (bodymindarts) [inline src/app/authentication/login.ts:373]: Why are we only passing the name on? Aren't we potentially silently dropping interesting error context?

File: src/graphql/error.ts [hunk 2]
```diff
@@ -365,6 +365,39 @@ export class InvalidPhoneForOnboardingError extends CustomApolloError {
...
+      code: "UNAUTHORIZED_VPN_IP_FOR_REWARD",
+      ...errData,
+    })
+  }
+}
+
+export class UnauthorizedIPForRewardError extends CustomApolloError {
+  constructor(errData: CustomApolloErrorData) {
+    super({
+      message: "This ip is not unauthorized for rewards.",
```
Comment (bodymindarts) [inline src/graphql/error.ts:393]: double negative? `not unauthorised`?

File: src/app/authentication/login.ts (hunk 1 above)
Comment (nicolasburtey) [reply src/app/authentication/login.ts:373]: yeah, this was the previous behavior. just changed it.

**Review by bodymindarts — APPROVED** [review]
LGTM

**Pull request merged**